### PR TITLE
cam6_3_108: Ocean emissions fix

### LINKED ISSUE #567: Pass lightning flash frequency to mediator for CTSM
See discussions in 
https://github.com/ESCOMP/CTSM/issues/1630

As suggested by @ekluzek, should set a flag in drv_flds_in so that surface models know that this is available.

### LINKED ISSUE #794: cam6_3_108:  Issue with ocean emissions chemistry code in B compsets
### What happened?

Floating point exception occurs in `src/chemistry/mozart/ocean_emis.F90` when configured with active ocean component (B compset).  Following is the traceback:
```
98:MPT: #2  MPI_SGI_stacktraceback (
98:MPT:     header=header@entry=0x7fff631f2410 "MPT ERROR: Rank 98(g:98) received signal SIGFPE(8).\n\tProcess ID: 46099, Host: r6i2n5, Program: /glade/scratch/fvitt/SMS_D.f09_g17.B1850.cheyenne_intel.allactive-defaultio.ocnemis/bld/cesm.exe\n\tMPT Ver"...) at sig.c:340
98:MPT: #3  0x00002ac95ca02e66 in first_arriver_handler (signo=signo@entry=8, 
98:MPT:     stack_trace_sem=stack_trace_sem@entry=0x2ac96c320080) at sig.c:489
98:MPT: #4  0x00002ac95ca030f3 in slave_sig_handler (signo=8, siginfo=<optimized out>, 
98:MPT:     extra=<optimized out>) at sig.c:565
98:MPT: #5  <signal handler called>
98:MPT: #6  0x0000000003e09ea6 in ocean_emis::henry_m_atm (henry_m_atm=..., 
98:MPT:     speciesindex=15, t_water_k=..., salinity_partsperthousand=..., ncol=16)
98:MPT:     at /glade/work/fvitt/cesm/cesm2_3_test/components/cam/src/chemistry/mozart/ocean_emis.F90:684
98:MPT: #7  0x0000000003e0342f in ocean_emis::flux_kg_m2_s (flux_kg_m2_s=..., 
98:MPT:     speciesname=..., cgas_ppt=..., cwater_nm=..., t_air_k=..., u10_m_s=..., 
98:MPT:     p_atm=..., t_water_k=..., salinity_partsperthousand=..., 
98:MPT:     switch_bubble=.FALSE., ncol=16, .tmp.SPECIESNAME.len_V$20a3=16)
98:MPT:     at /glade/work/fvitt/cesm/cesm2_3_test/components/cam/src/chemistry/mozart/ocean_emis.F90:478
98:MPT: #8  0x0000000003dfd5de in ocean_emis::ocean_emis_getflux (lchnk=1165, ncol=16, 
98:MPT:     state=..., u10=..., sst=..., ocnfrac=..., icefrac=..., sflx=...)
98:MPT:     at /glade/work/fvitt/cesm/cesm2_3_test/components/cam/src/chemistry/mozart/ocean_emis.F90:327
98:MPT: #9  0x0000000002b06576 in chemistry::chem_emissions (state=..., cam_in=...)
98:MPT:     at /glade/work/fvitt/cesm/cesm2_3_test/components/cam/src/chemistry/mozart/chemistry.F90:895
98:MPT: #10 0x000000000124d1f6 in physpkg::tphysac (ztodt=1800, cam_in=..., 
98:MPT:     cam_out=..., state=..., tend=..., pbuf=0x2ac975ae7340)
98:MPT:     at /glade/work/fvitt/cesm/cesm2_3_test/components/cam/src/physics/cam/physpkg.F90:1509
```

### What are the steps to reproduce the bug?

```
./create_test --walltime 00:30:00 SMS_D.f09_g17.B1850.cheyenne_intel.allactive-defaultio
```

### What CAM tag were you using?

cam6_3_106

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

/glade/scratch/fvitt/SMS_D.f09_g17.B1850.cheyenne_intel.allactive-defaultio.ocnemis

### Will you be addressing this bug yourself?

Yes

### Extra info

_No response_

### PR DESCRIPTION
**Pull request by fvitt**
Fix issue with ocean emissions in B compsets where SSTs are zero over land.

fixes #794 
closes #567
closes #747

### CODE REVIEW

**Review by jedwards4b — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]

File: src/chemistry/mozart/ocean_emis.F90 [hunk 1]
```diff
@@ -235,35 +236,39 @@ subroutine ocean_emis_init()
...
+       call lininterp(wrk2d, file_nlon, file_nlat, salinity(1:ncols,c), ncols, lon_wgts, lat_wgts)
 
        call lininterp_finish(lon_wgts)
        call lininterp_finish(lat_wgts)
 
     end do
 
+    where(salinity < 0._r8)
+       salinity = 33.0_r8
+    end where
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/chemistry/mozart/ocean_emis.F90:248]: Good question.  I don't know.  This was moved from line 475.  I am preserving bit-for-bit answers here.

[the remark this answers is not in the document]
Comment (cacraigucar) [reply src/chemistry/mozart/ocean_emis.F90:248]: @fvitt - Doing a "blame", line 475 was something you brought in along with Siyaun Wang back in April 2020.  Could you please reach out to Siyaun and check whether this is something which should be changed, and if so, please open an issue.  I will approve this PR "as-is" to preserve answers for this PR.

**Review by cacraigucar — APPROVED** [review]
Approving with a caveat that the salinity value for when salinity is less than zero might need to be changed in a future update.

File: src/chemistry/mozart/ocean_emis.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (fvitt) [reply src/chemistry/mozart/ocean_emis.F90:248]: This is quick explanation from Siyaun:
"In general 33 is a very typical ballpark value for salinity, and overall the air-sea exchange process is not really sensitive to salinity, and that's why I'm using climatology for modern-day simulations.  There might be a few grids with bad salinity values but are good in other regards and I wanted to save those grids."

So I don't think we should change it.  It is filling in missing values with a typical value of 33.

**Pull request merged**